random.js

```diff
@@ -0,0 +1,76 @@
+var slideIndex = 0;
+var slides = document.getElementsByClassName("mySlides");
+var dots = document.getElementsByClassName("slide-inactive");
+var i;
```

Comment:
Why declare this at the file scope?

Reply:
I declared it at the function scope because both of the functions (showSlides) and (slideAutomatic) require it. I figure this will just make it global and easily accessible by both.

If theres a better solution, I'm curious.

Reply:
Unless they actually share the same data, they should both declare their own variable. The performance overhead of allocating 32 bits for a new `let` is worth the code quality.

Reply:
Very interesting. Thank you.